### ISSUE TITLE
Ability to mark a facility as closed/shutdown

### ISSUE THREAD

**Issue by as44980** [peeringdb/peeringdb#464]
Ideally this could replace the current practice of indicating this with free text in the facility name, as somebody appears to have done for multiple locations... ;)

Setting the 'facility shutdown' flag could least prevent new newtworks or IX's from indicating they are present at the facility, and potentially for the operator to remove these net/ix records while leaving the fac record for historical reference...

**Comment by arnoldnipper** [comment]
Is there any reason to still show it when it is closed/shutdown? Why not simply remove it?

**Comment by mcmanuss8** [comment]
@arnoldnipper  I know companies will put pdb record ids in various systems. It'd be nice if, when pulling up that object, we saw that it was marked as deconned/shutdown versus a 404. In the later case you might think someone put a bad external key, in the former you can verify this was what you thought it was, but it's no longer relevant. Seems saner to me, and database rows are cheap. :)

**Comment by arnoldnipper** [comment]
Same happens already when records are merged. We shouldn't make the DB too complicated imho just to cover corner cases

**Comment by arnoldnipper** [comment]
Maybe implemting #451 would already solve the issue

**Comment by as44980** [comment]
I had envisaged the records remaining visible in the search results, particularly to help with the _i'm sure company X had a facility in metro Y_ type questions..

However is this is considered too much of a corner case perhaps the scope could be broadened to an "Is Operational" flag which is enabled by default, but can be deselected by the record owner to cover the use cases of wishing to list the facility before it has been declared operational or to indicate that it is no longer operational as data center...

In which case it would probably not be desirable for deselecting this option to remove any net/ix records, it does seem reasonable that if the facility is not operational then it should not be possible for any new records to be created claiming to be located there :)

**Comment by arnoldnipper** [comment]
> I had envisaged the records remaining visible in the search results, particularly to help with the _i'm sure company X had a facility in metro Y_ type questions..

You still would be able when querying via the API. I can't see a need to have this in the GUI and hence adding an extra flag. Same also would hold true for IXes and maybe also for nets (which we indeed could as we are versioning).

Honestly I don't see any **added operational value** having this flag

**Comment by as44980** [comment]
So is the recommendation that operators should request the facility record be deleted, and if so how soon after the facility is considered to have been shutdown?

And was there any conclusion to #451 ?  :)

**Comment by arnoldnipper** [comment]
That would be my recommendation. And it's up to the operator when to notify us. But we may also do if we  get notified by a 3rd party

**Comment by arnoldnipper** [comment]
> And was there any conclusion to #451 ? :)

yes, see #451

**Comment by grizz** [comment]
> @arnoldnipper I know companies will put pdb record ids in various systems. It'd be nice if, when pulling up that object, we saw that it was marked as deconned/shutdown versus a 404. In the later case you might think someone put a bad external key, in the former you can verify this was what you thought it was, but it's no longer relevant. Seems saner to me, and database rows are cheap. :)

I like this as a use case, when someone goes to an ID that would 404, it can show a Not Found page and then reference the deleted record if it exists. Something like "Were you looking for this? It has been deleted."

@peeringdb/pc votes please!

**Comment by grizz** [comment]
@as44980 thanks for bringing this back up, you're very correct this ties into #696 which is slated for development soon.

@peeringdb/pc what's the problem?

**Comment by arnoldnipper** [comment]
-1 as there you always may search for _closed_ facilities, because they are still in the DB. Just specify status=deleted

**Comment by as44980** [comment]
> -1 as you always may search for _closed_ facilities, because they are still in the DB. Just specify status=deleted

While the invisible status codes may be useful to some, IMHO there is value in at least the fac records remaining visible and public even after the cease to be operated as datacentres.. if for nothing else than to provide a somewhat authoritative source of historical data :)

However this does make me wonder if this existing "status" could be expanded upon, perhaps scope creep but is there any value in considering the actual life-cycle of these facilities?

ie allow the fac owner to select status from a list such as:

Under Construction
Operational
Decomissioned

**Comment by arnoldnipper** [comment]
> I like this as a use case, when someone goes to an ID that would 404, it can show a Not Found page and then reference the deleted record if it exists. Something like "Were you looking for this? It has been deleted."

Could we extend this to all objects? At least `org`, `ix`, `fac` and `net`? If so, +1

**Comment by mcmanuss8** [comment]
> Could we extend this to all objects? At least org, ix, fac and net? If so, +1

I think it should apply to all objects, yes. +1 from me for that

**Comment by job** [comment]
Facilities that are shut down or otherwise no longer available to interconnection stakeholders should just be deleted. The 404 is a correct response, the facility no longer exists. By deleting the object networks and IXPs can't connect to it any longer, which is the intended outcome as the facility has been deleted.

**Comment by leovegoda** [comment]
The Product Committee has decided to close this issue. There was no agreement on the concept and the issue has not been discussed for six months or longer.

**Comment by as44980** [comment]
Shame, I guess the DB is going to remain populated by a number of fac records for facilities which have been shutdown :(

**Comment by martinhannigan** [comment]
Can you give us an example?

**Comment by as44980** [comment]
> Can you give us an example?

https://www.peeringdb.com/search?q=no%20longer%20operational

**Comment by martinhannigan** [comment]
This is a good issue. It seems clear at least with the examples that its a
corner case. However, accuracy is a goal. The problem also ties back to the
conversation we're having about addresses vs. facilities=FAC's. Is the
building no longer supporting data centers or interconnection activity or
did a single FAC shut down which happen to gain control of the address? In
the EQIX cases, I don't know if they were acquired or changed hands. Did
they actually empty out a whole building and no one provides any DC
services any longer?

Not disagreeing with you, but want to shake out the whole issue a little
better. I'm willing to shephard something more clear and precise back to
the PC.

Thanks,

Martin

**Comment by grizz** [comment]
We just need a clear feature spec for this and I think we'd all +1 it. :) Probably just status=deleted, and if the record is referenced a bit better message.

**Comment by martinhannigan** [comment]
It may be as simple as marking "Closed" or "Reported Closed". How do we do
this and minimize work (and errors)?

I researched some of these today. While we mark them closed, should we also
remove them from search? They are 'dirty', they still have networks
appearing present and show up in other searches. See
https://www.peeringdb.com/ix/60 as an example.

**Comment by grizz** [comment]
Simplest is set status = "closed" with a validation check that fails if any entity still lists itself as present there next time they edit.

It's also a good litmus test for stale data I guess. :)

**Comment by martinhannigan** [comment]
User reports it and it auto closes? Then wiping the stale data is a human
process?

**Comment by martinhannigan** [comment]
Entity check addresses what I called "dirty". Should we still mark it as
closed even if its dirty, but not wipe it? If its really closed it would
still be closed.

**Comment by grizz** [comment]
From the previous discussion, I think it makes more sense to leave it around and just have it marked as status closed.

Also, I would think the AC should approve close requests, I can't imagine they happen very often.

**Comment by martinhannigan** [comment]
On Mon, Jun 20, 2022 at 12:55 PM Matt Griswold ***@***.***>
wrote:

> From the previous discussion, I think it makes more sense to leave it
> around and just have it marked as status closed.
>
> Also, I would think the AC should approve close requests, I can't imagine
> they happen very often.
>
> If we mark a facility closed, would we simply remove all the networks that
failed to de-link? I dont really care about the dirty data; one will know
that if the FAC is closed the records are "lame".

**Comment by grizz** [comment]
Yeah, we can just leave them and since they fail validation, the network will have to remove it next time they edit.

**Comment by martinhannigan** [comment]
On Mon, Jun 20, 2022 at 1:14 PM Matt Griswold ***@***.***>
wrote:

> Yeah, we can just leave them and since they fail validation, the network
> will have to remove it next time they edit.
>
>
I think this is done too. +1

**Comment by martinhannigan** [comment]
Do you know of any that are closed but no indication of such?

**Comment by as44980** [comment]
> Do you know of any that are closed but no indication of such?

"SloughIX" (aka Network-I) may be a candidate:

https://www.peeringdb.com/fac/571

**Comment by as44980** [comment]
> Is the building no longer supporting data centers or interconnection activity or did a single FAC shut down which happen to gain control of the address? In the EQIX cases, I don't know if they were acquired or changed hands. Did they actually empty out a whole building and no one provides any DC services any longer? 

The last few years have seen a number of facilities closed following the expiration of their leases and, particularly where these occupied one or more floors of a mixed use building, it seems to be common that the landlords seek more conventional tenants and/or take the opportunity to redevelop the building :)

And while I suspect that all of the examples with the string "No longer operational" in the facility name were stripped back to the original pre-lease condition prior to being handed back to the landlord, it would seem plausible that a facility which shutdown due to bankruptcy of the operator may be acquired and re-opened by another operator :)
